react-app/src/redux/store.js

```diff
@@ -1,11 +1,27 @@
-import { combineReducers, createStore } from 'redux';
+import { combineReducers, createStore, applyMiddleware } from 'redux';
+import { reducer as form } from 'redux-form';
+import thunk from 'redux-thunk';
 
+import { saveState } from './localStorage';
+import authReducer from './Auth/reducer';
 import gameReducer from './Game/reducer';
 
 const rootReducer = combineReducers({
-  game: gameReducer
+  game: gameReducer,
+  auth: authReducer,
+  form
 });
 
-const store = createStore(rootReducer);
+const store = createStore(
+  rootReducer,
+  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(), // eslint-disable-line no-underscore-dangle
+  applyMiddleware(thunk)
+);
+
+const saveToLocalStorage = () => {
+  saveState('auth', store.getState().auth);
```

Comment:
Are you sure we really want to save the whole state in the local storage. Notice that some user data may have changed and we arent making a new request to the backend. Maybe you can save just the required data in order to authenticate the user